snap/backend/src/utils/accountUtils.js

```diff
@@ -179,6 +197,75 @@ const accountUtils = {
 		} catch (error) {
 			throw new Error(`Failed to import account: ${error.message}`);
 		}
+	},
+	/**
+	 * Fetch account mosaic and update mosaic info.
+	 * @param {{ network: object }} state - The snap state object.
+	 * @param {Array<string>} addresses - The account addresses.
+	 * @returns {Promise<Record<string, Array<AccountMosaics>>>} - The accounts mosaics object.
+	 */
+	async fetchAndUpdateAccountMosaics(state, addresses) {
+		const { network } = state;
+
+		const client = symbolClient.create(network.url);
+		const accountsMosaics = await client.fetchAccountsMosaics(addresses);
+
+		// Get all unique mosaic ids
+		const mosaicIds = [...new Set(Object.values(accountsMosaics).flatMap(accountMosaics => accountMosaics.map(mosaic => mosaic.id)))];
```

Comment:
I think the mosaics list for an account is always unique.

Reply:
yes, you are right.
But in this case, I will collect all the mosaic IDs from all accounts.

For example, I have 10 accounts, and each account will have a mosaics list.
I will collect all mosaic IDs, and filter into unique mosaic ids.

use the id to query mosaic info.